From 8c1f9e467264f40fb8e73789a0a896a930212aa4 Mon Sep 17 00:00:00 2001
From: Matt Selsky <matthew.selsky@twosigma.com>
Date: Sat, 26 Nov 2016 10:59:15 -0500
Subject: [PATCH 024/268] Fix mrulist sorting

dict's use get, not getkey
fix exception name for bad sort keys
keep sort key and sort function variables distinct
---
 pylib/packet.py | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/pylib/packet.py b/pylib/packet.py
index 87dd420..02510ca 100644
--- a/pylib/packet.py
+++ b/pylib/packet.py
@@ -1169,7 +1169,7 @@ class ControlSession:
 
         if variables:
             if "sort" in variables:
-                sorter = variables["sort"]
+                sortkey = variables["sort"]
                 del variables["sort"]
                 # FIXME: implement sorting by address, in case anyone cares
                 sortdict = {
@@ -1182,12 +1182,12 @@ class ControlSession:
                 "count" : lambda e: e.ct,	# hit count ascending
                 "-count": lambda e: -e.ct,	# hit count descending
                 }
-                if sorter == "listint":
-                    sorter = None
-                if sorter is not None:
-                    sorter = sortdict.getkey(key)
+                if sortkey == "listint":
+                    sortkey = None
+                if sortkey is not None:
+                    sorter = sortdict.get(sortkey)
                     if sorter == None:
-                        raise ControlException(BAD_SORT % key)
+                        raise ControlException(SERR_BADSORT % sortkey)
             for k in list(variables.keys()):
                 if k in ("mincount", "resall", "resany",
                          "maxlstint", "laddr", "sort"):
-- 
2.7.4

